File: JavaToKotlin/apps(kotlin)/src/main/kotlin/io/dustin/apps/board/domain/qna/question/model/dto/QuestionDto.java
package io.dustin.apps.board.domain.qna.question.model.dto;

import io.dustin.apps.board.domain.qna.question.model.Question;
import io.dustin.apps.common.model.IdAble;
import lombok.Builder;

import java.time.LocalDateTime;

@Builder
public record QuestionDto(
        Long id,

        Long userId,

        String subject,

        String content,

        Boolean isComment,

        LocalDateTime createdAt
) implements IdAble {
        public static QuestionDto from(Question question) {
            return QuestionDto.builder()
                    .id(question.getId())
                    .userId(question.getUserId())
                    .subject(question.getSubject())
                    .content(question.getContent())
                    .createdAt(question.getCreatedAt())
                    .build();
        }

    @Override
    public Long getId() {
        return this.id;
    }

}

File: apps/src/main/kotlin/io/dustin/apps/board/domain/qna/question/model/dto/QuestionDto.kt
package io.dustin.apps.board.domain.qna.question.model.dto

import com.fasterxml.jackson.annotation.JsonFormat
import io.dustin.apps.board.domain.qna.question.model.Question
import io.dustin.apps.common.model.IdAble
import io.swagger.v3.oas.annotations.media.Schema
import java.time.LocalDateTime

/**
 * 1:1 문의 질문 dto
 */
@Schema(description = "1:1 문의 질문 관련 응답 객체")
data class QuestionDto(
    @Schema(description = "id")
    val id: Long,

    @Schema(description = "질문을 작성한 유저 id")
    val userId: Long,

    @Schema(description = "질문 제목")
    var subject: String,

    @Schema(description = "질문 내용")
    var content: String,

    @Schema(description = "답변 여부")
    var isAnswer: Boolean,

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @Schema(type = "string", description = "객체 생성 날짜", example = "1900-01-01 23:59:59")
    val createdAt: LocalDateTime
) : IdAble {


    companion object {
        fun from(question: Question) = with(question) {
            QuestionDto(
                id = id ?: 0L,
                userId = userId,
                subject = subject,
                content = content,
                isAnswer = false,
                createdAt = createdAt

            )
        }
    }
    override fun id(): Long = id
}
